lib/Conversion/TorchToStablehlo/Basic.cpp

```diff
@@ -931,10 +931,10 @@ LogicalResult ConvertAtenOp<AtenBatchNormOp>::matchAndRewrite(
   Value momentum = adaptor.getMomentum();
   (void)momentum;
 
-  if (inputTy.getRank() <= 2) {
-    return rewriter.notifyMatchFailure(op,
-                                       "input should have rank larger than 2");
-  }
+  // handle feature index, see torch's BatchNorm1d, BatchNorm2d, BatchNorm3d,
+  // all of NC, NCL, NCHW, NCDHW's feature index is 1.
+  int64_t feature_index = 1;
```

Comment:
Pardon my ignorance, but could you elaborate why the feature index is always 1?

Reply:
You could see torch's documents:
https://pytorch.org/docs/stable/generated/torch.nn.functional.batch_norm.html?highlight=batch_norm#torch.nn.functional.batch_norm
https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html#torch.nn.BatchNorm1d

PyTorch has BatchNorm1d/2d/3d. BatchNorm1d has layout NC and NCL, BatchNorm2d has layout NCHW, BatchNorm3d has layout NCDHW. All of them have the C dim at index 1. So the feature index is always 1.

Reply:
BTW, I didn't see any description about Channel Last format in batch norm.

Reply:
Ah, I see.  Thanks for explaining!  It makes sense now.